@examples//matrix:native_matrix fails to build

The for loops in examples/matrix/src/matrix.c don't compile because they require C99 mode:

```
$ bazel build @examples//matrix:native_matrix
INFO: Found 1 target...
ERROR: /home/jcater/.cache/bazel/_bazel_jcater/f161603a652c64478ce40dd2b7d0ce80/external/examples/matrix/BUILD:5:1: C++ compilation of rule '@examples//matrix:native_matrix' failed: Process exited with status 1 [sandboxed].
external/examples/matrix/src/matrix.c: In function 'matrix_transpose':
external/examples/matrix/src/matrix.c:73:3: error: 'for' loop initial declarations are only allowed in C99 mode
   for (size_t i = 0; i < q; ++i) {
   ^
external/examples/matrix/src/matrix.c:73:3: note: use option -std=c99 or -std=gnu99 to compile your code
external/examples/matrix/src/matrix.c: In function 'matrix_equal':
external/examples/matrix/src/matrix.c:107:3: error: 'for' loop initial declarations are only allowed in C99 mode
   for (size_t i = 0; i < len; ++i) {
   ^
Use --strategy=CppCompile=standalone to disable sandboxing for the failing actions.
Target @examples//matrix:native_matrix failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 0.274s, Critical Path: 0.11s
```

--- a/examples/matrix/BUILD
+++ b/examples/matrix/BUILD
@@ -6,11 +6,13 @@ cc_library(
     name = "native_matrix",
     srcs = ["src/matrix.c"],
     hdrs = ["src/matrix.h"],
+    copts = ["-std=c99"],
 )
 
 cc_test(
     name = "native_matrix_test",
     srcs = ["src/matrix_test.c"],
+    copts = ["-std=c99"],
     deps = [
         ":native_matrix",
     ],
@@ -25,7 +27,7 @@ rust_library(
     ],
     deps = [
         ":native_matrix",
-        "@libc//:libc"
+        "@libc//:libc",
     ],
 )
 

